From 09369fc0380a164974de7efb319e899399c696fa Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 2 May 2014 14:36:36 -0400
Subject: [misc] mei: make sure that me_clients_map big enough before copying

Message-id: <1399041470-21330-4-git-send-email-prarit@redhat.com>
Patchwork-id: 79291
O-Subject: [RHEL7.1 PATCH BZ 1088953 03/77] mei: make sure that me_clients_map big enough before copying
Bugzilla: 1088953
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088953

commit 23f5a322063cba7c9f5dba5205e2960a24291373
Author: Tomas Winkler <tomas.winkler@intel.com>
Date:   Mon Sep 2 03:11:01 2013 +0300

    mei: make sure that me_clients_map big enough before copying

    To make static analyzers happy validated that
    sizeof me_clients_map  is larger than sizeof valid_addresses from the
    enumeration response before memcpy
    We can use BUILD_ON macro as both arrays are defined statically

    Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/mei/hbm.c b/drivers/misc/mei/hbm.c
index 5a5ae07..cdd0dff 100644
--- a/drivers/misc/mei/hbm.c
+++ b/drivers/misc/mei/hbm.c
@@ -677,7 +677,10 @@ void mei_hbm_dispatch(struct mei_device *dev, struct mei_msg_hdr *hdr)
 
  case HOST_ENUM_RES_CMD:
   enum_res = (struct hbm_host_enum_response *) mei_msg;
-  memcpy(dev->me_clients_map, enum_res->valid_addresses, 32);
+  BUILD_BUG_ON(sizeof(dev->me_clients_map)
+    < sizeof(enum_res->valid_addresses));
+  memcpy(dev->me_clients_map, enum_res->valid_addresses,
+   sizeof(enum_res->valid_addresses));
   if (dev->dev_state == MEI_DEV_INIT_CLIENTS &&
       dev->hbm_state == MEI_HBM_ENUM_CLIENTS) {
     dev->init_clients_timer = 0;
-- 
1.7.1